modules/instances/src/main/java/net/hollowcube/world/BaseWorld.java

```diff
@@ -0,0 +1,134 @@
+package net.hollowcube.world;
+
+import net.hollowcube.world.compression.WorldCompressor;
+import net.hollowcube.world.decompression.WorldDecompressor;
+import net.hollowcube.world.dimension.DimensionTypes;
+import net.hollowcube.world.event.PlayerInstanceLeaveEvent;
+import net.hollowcube.world.event.PlayerSpawnInInstanceEvent;
+import net.hollowcube.world.util.FileUtil;
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.entity.Player;
+import net.minestom.server.event.EventDispatcher;
+import net.minestom.server.event.instance.RemoveEntityFromInstanceEvent;
+import net.minestom.server.event.player.PlayerSpawnEvent;
+import net.minestom.server.instance.AnvilLoader;
+import net.minestom.server.instance.Instance;
+import net.minestom.server.instance.InstanceContainer;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ForkJoinPool;
+
+@SuppressWarnings("UnstableApiUsage")
+public class BaseWorld implements World {
```

Comment:
What will be the scope of usage for a base world? Is it essentially what we will base all worlds off of (for map maker at least)? Also what will be the limit of using WORLD_DIR to keep maps in memory rather than updating in S3 and pulling later? Will that be entirely up to the dev using the common package?